prov/shm/src/smr_util.c

```diff
@@ -284,6 +286,13 @@ int smr_create(const struct fi_provider *prov, struct smr_map *map,
 
 	(*smr)->cma_cap_peer = SMR_VMA_CAP_NA;
 	(*smr)->cma_cap_self = SMR_VMA_CAP_NA;
+	(*smr)->xpmem_cap_self = (xpmem) ? SMR_VMA_CAP_ON :
+		SMR_VMA_CAP_OFF;
+	if (xpmem)
+		(*smr)->xpmem_self = xpmem->pinfo;
+	else
+		memset(&(*smr)->xpmem_self, 0,
+		       sizeof((*smr)->xpmem_self));
```

Comment:
Can you refactor this to be
```
if (xpmem) {
    //set xpmem cap and info (and maybe move env var check to here too)
}
```
do we even need the else if the region initializes to all 0s?